src/interfaces.jl

```diff
@@ -398,12 +487,45 @@ eval(default_plot_signatures(:plot, :plot!, :Any))
 
 # plots to scene
 
+plotfunc(::Combined{F}) where F = F
+
+
+
 """
 Main plotting signatures that plot/plot! route to if no Plot Type is given
 """
-function plot!(scene::SceneLike, ::Type{Any}, attributes::Attributes, args...)
-    PlotType = plottype(to_value.(args)...)
-    plot!(scene, PlotType, attributes, args...)
+function plot!(scene::SceneLike, P::PlotFunc, attributes::Attributes, args...; kw_attributes...)
+    attributes = merge!(Attributes(kw_attributes), attributes)
+    argvalues = to_value.(args)
+    PreType = plottype(P, argvalues...)
+    # plottype will lose the argument types, so we just extract the plot func
+    # type and recreate the type with the argument type
+    PreType = Combined{plotfunc(PreType), typeof(argvalues)}
+    keys = used_attributes(PreType, argvalues...)
```

Comment:
maybe `keys = intersect(used_attributes(PreType, argvalues...), keys(attributes))` ? Otherwise we throw unless the user provides all keywords

Reply:
keys should be immutable ;) so we'd need to filter them out a step later!

Reply:
ah, but the number of passed attributes  wont change either^^ so, guess that should work :)